Can't create custom analyzers with 7.1.0

I have a pretty complex custom analyzer that can't be created with pretty weird error message:

> Caused by: java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:264)
	at com.sksamuel.elastic4s.XContentBuilder.endObject(XContentBuilder.scala:336)
	at com.sksamuel.elastic4s.requests.analysis.MappingCharFilter.build(charFilter.scala:19)
	at com.sksamuel.elastic4s.requests.analysis.AnalysisBuilder$.$anonfun$build$4(Analysis.scala:36)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at com.sksamuel.elastic4s.requests.analysis.AnalysisBuilder$.build(Analysis.scala:36)
	at com.sksamuel.elastic4s.requests.indexes.CreateIndexContentBuilder$.$anonfun$apply$3(CreateIndexContentBuilder.scala:32)
	at scala.Option.foreach(Option.scala:257)
	at com.sksamuel.elastic4s.requests.indexes.CreateIndexContentBuilder$.apply(CreateIndexContentBuilder.scala:32)
	at com.sksamuel.elastic4s.requests.indexes.admin.IndexAdminHandlers$CreateIndexHandler$.build(IndexAdminHandlers.scala:194)

Here is the analyzer creation code:
`

    val stopWordsFilter = StopTokenFilter("stop", stopwords = List("a", "b") ignoreCase = Some(true))

    val triplets = NGramTokenizer("triplets", minGram=3, maxGram=3, tokenChars=Seq("letter", "digit"))

    val mapping = MappingCharFilter("mapping", Map("a" -> "b"))

    val wordDelimiter = WordDelimiterTokenFilter("delimiter",
      generateWordParts = Some(true),
      generateNumberParts = Some(true),
      catenateWords = Some(true),
      catenateNumbers = Some(false),
      catenateAll = Some(true),
      splitOnCaseChange = Some(true),
      preserveOriginal = Some(true),
      splitOnNumerics = Some(false)
    )

    val language = HunspellTokenFilter(name = "hunspell", locale = "ru")

    val tx= CustomAnalyzer("tx", "triplets", List("mapping"), List("stop", "delimiter", "hunspell"))

    val analysis = Analysis(List(tx), List(triplets), List(stopWordsFilter, wordDelimiter, language), List(mapping))

    client.execute {
      createIndex(INDEX_NAME)
        .mapping(
          properties(Seq(
            TextField("tx").analyzer("tx")
          ))
        )
        .analysis(analysis)
    }`

What i'm doing wrong?

--- a/elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/analysis/charFilter.scala
+++ b/elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/analysis/charFilter.scala
@@ -16,7 +16,7 @@ case class MappingCharFilter(override val name: String,
     mappings.foreach { case (k, v) =>
       b.value(s"$k=>$v")
     }
-    b.endObject()
+    b.endArray()
     b.endObject()
   }
 }

